proto/storage/cluster.proto

```diff
@@ -94,6 +95,43 @@ message AdmissionControllerConfig {
   bool enforce_on_updates = 5;
 }
 
+message ProcessConfig {
+  bool enabled = 1;
+}
+
+message NetworkConnectionConfig {
+  bool enabled = 1;
+  bool aggregate_external = 2;
+}
+
+message NetworkEndpointConfig {
+  bool enabled = 1;
+  bool include_listening_endpoint_processes = 2;
+}
+
+message CollectorFeature {
+  oneof feature {
+    ProcessConfig processes = 1;
+    NetworkConnectionConfig network_connections = 2;
+    NetworkEndpointConfig network_endpoints = 3;
+  }
+}
+
+message CollectorNamespaceConfig {
+  message NamespaceRule {
+    string namespace = 1;
+    storage.MatchType match_type = 2;
+  }
+
+  CollectorFeature feature = 1;
+  repeated NamespaceRule namespace_selection = 2;
+}
+
+message CollectorConfig {
+  repeated CollectorFeature cluster_scope_config = 1;
+  repeated CollectorNamespaceConfig namespace_scope_config = 2;
```

Comment:
IMO this is pretty confusing.  A namespace config can only have one feature.  But the cluster can have a list of features.  Could the same namespace rules not apply to multiple features at once?  

The change I'm asking for here is something somewhere that explains this setup so that the next person that looks at it can easily understand the structure and why it is.  So some comments maybe?

Reply:
I have added comments that hopefully explain the structure. A namespace config could have an array of type CollectorFeature, but that would be more complicated. How that concept would be displayed in the UI is not clear to me. In what is displayed here cluster_scope_config is an array of type CollectorFeature. I have changed this based on @SimonBaeumer's comment above, such that each CollectorFeature type is hard coded instead.